lib/easy.c

```diff
@@ -182,10 +182,12 @@ static CURLcode global_init(long flags, bool memoryfuncs)
   }
 #endif
 
+#ifdef __APPLE__
   if(Curl_macos_init()) {
     DEBUGF(fprintf(stderr, "Error: Curl_macos_init failed\n"));
     goto fail;
   }
+#endif
```

Comment:
The idea is to provide an empty macro for `Curl_macos_init()` when not needed so that this won't need an #ifdef...

Reply:
I do realize that, but I was looking at the two blocks above this one, where `Curl_win32_init` is only called when `WIN32` is defined and `Curl_amiga_init` is only called when `__AMIGA__` is defined so I was trying to do something similar for `Curl_macos_init`. But I can remove the `#ifdef __APPLE__`…`#endif` again if you prefer.

Reply:
Yes please, I will get rid of the other ones too: #11525